src/uu/ls/src/ls.rs

```diff
@@ -2401,6 +2464,32 @@ fn display_file_name(
     }
 }
 
+fn hyperlink_name(hyperlink_config: &mut HyperlinkConfig, path: &Path, name: String) -> String {
```

Comment:
With that change, you can probably also remove `HyperLinkConfig` and just use `HyperLinkData`.

Reply:
That would either require to non-lazily initialize the data, which would do unnecessary work in case of empty directories or no relative paths, or alternatively to use a lazy_static!.

Would a lazy_static! be better? I didn't like it that much since it's global and in principle the hyperlink data might eventually depend on other options (even though it currently doesn't)

The other option is to create a struct that contains the Config plus mutable data and pass that around instead of Config, but not sure if that's worth the added code.

Reply:
Alternatively could also use RefCell, but that adds unnecessary runtime checks.

Reply:
Initializing the data only happens once. I'd be really surprised if you were able to measure a meaningful difference. *If* you do, then  you could look into using a [oncecell](https://docs.rs/once_cell/1.7.2/once_cell/index.html), which fits this kind of thing very well and works with `&` references if I'm not mistaken.